src/www/ui/api/Controllers/JobController.php

```diff
@@ -167,6 +167,69 @@ public function createJob($request, $response, $args)
     $error = new Info(400, "Folder id and upload id should be integers!", InfoType::ERROR);
     return $response->withJson($error->getArray(), $error->getCode());
   }
+
+  /**
+   * Delete a job using it's Job ID and Queue ID. Job ID is job_pk in job table
+   * and Queue ID is jobqueue_pk in jobqueue table
+   * 
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function deleteJob($request, $response, $args)
+  {
+    /* Check if the job exists */
+    $jobId  = intval($args['id']);
+    if (! $this->dbHelper->doesIdExist("job", "job_pk", $jobId)) {
+      $returnVal = new Info(404, "Job id " . $jobId . " doesn't exist",
+        InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+    
+    /* Check if user has permission to delete this job*/
+    $canDeleteJob = $this->restHelper->getJobDao()->hasActionPermissionsOnJob($jobId, Auth::getUserId(), Auth::getGroupId());
+    if (! $canDeleteJob) {
+      $returnVal = new Info(403, "You don't have permission to delete this job.",
+        InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    $queueId = $args['queue']; 
+
+    /* Get Jobs that depend on the job to be deleted */
+    $JobQueue = $this->restHelper->getShowJobDao()->getJobInfo([$jobId])[$jobId]["jobqueue"];
+
+    $dependentJobs = [];
+
+    if (array_key_exists($queueId, $JobQueue)) {
+      array_push($dependentJobs, $queueId);
+    } else {
+      $returnVal = new Info(404, "Job queue " . $queueId . " doesn't exist in Job " . $jobId,
+        InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    foreach ($JobQueue as $job) {
+      if (in_array($queueId, $job["depends"])) {
+       array_push($dependentJobs, $job["jq_pk"]);
+      }
+    }
+
+    /* Delete All jobs in dependentJobs */
+    foreach ($dependentJobs as $job) {
+      $Msg = "\"" . _("Killed by") . " " . $_SESSION[AUTH::USER_NAME] . "\"";
```

Comment:
Please avoid using magic variable `$_SESSION` in API. Use following functions instead to get user object.

$userId = $this->restHelper->getUserId();
$userName = $this->restHelper->getUserDao()->getUserName($userId);